sdk/spring/azure-spring-boot-starter-active-directory/README.md

```diff
@@ -0,0 +1,223 @@
+* Authenticate in frontend, sends bearer authorization code to backend, in backend a Spring Security filter validates the Jwt token from Azure AD and save authentication. The Jwt token is also used to acquire a On-Behalf-Of token for Azure AD Graph API so that authenticated user's membership information is available for authorization of access of API resources. Below is a diagram that shows the layers and typical flow for Single Page Application with Spring Boot web API backend that uses the filter for Authentication and Authorization.
+![Single Page Application + Spring Boot Web API + Azure AD](resource/spring-aad.png)
+
+The authorization flow is composed of 3 phrases:
+* Login with credentials and validate id_token from Azure AD 
+* Get On-Behalf-Of token and membership info from Azure AD Graph API
+* Evaluate the permission based on membership info to grant or deny access
+
+##Getting started
+#### Register the Application in Azure AD
+* **Register a new application**: Go to Azure Portal - Azure Active Directory - App registrations - New application registration to register the application in Azure Active Directory.  `Application ID` is `client-id` in `application.properties`.
+* **Grant permissions to the application**: After application registration succeeded, go to API ACCESS - Required permissions - DELEGATED PERMISSIONS, tick `Access the directory as the signed-in user` and `Sign in and read user profile`. Click `Grant Permissions` (Note: you will need administrator privilege to grant permission).
+* **Create a client secret key for the application**: Go to API ACCESS - Keys to create a secret key (`client-secret`).
+
+#### Add Maven Dependency
+
+`azure-spring-boot-starter-active-directory` is published on Maven Central Repository.
+If you are using Maven, add the following dependency.
+
+[//]: # "{x-version-update-start;com.azure:azure-spring-boot-starter-active-directory;dependency}"
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-active-directory-spring-boot-starter</artifactId>
+    <version>2.2.5-beta.1</version>
+</dependency>
+```
+[//]: # "{x-version-update-end}"
+
+## Examples
+#### Configure application.properties and autowire beans
+
+Refer to different samples for different authentication ways. 
+
+##### Authenticate in backend
+
+Please refer to [azure-spring-boot-sample-active-directory-backend](../azure-spring-boot-samples/azure-spring-boot-sample-active-directory-backend/README-a.md) for authenticate in backend.
+
+Configure application.properties:
+```properties
+spring.security.oauth2.client.registration.azure.client-id=xxxxxx-your-client-id-xxxxxx
+spring.security.oauth2.client.registration.azure.client-secret=xxxxxx-your-client-secret-xxxxxx
+
+azure.activedirectory.tenant-id=xxxxxx-your-tenant-id-xxxxxx
+azure.activedirectory.active-directory-groups=group1, group2
+```
+
+Autowire `OAuth2UserService` bean in `WebSecurityConfigurerAdapter`:
+```java
```

Comment:
For code samples in README, we use a tool called embedme to inject codesnippets. Consider using that so we can be sure that the code samples in readmes are compilable.

Reply:
done